src/Internal/Wallet/Cip30.js

```diff
@@ -5,16 +5,29 @@ exports._getNetworkId = conn => () => conn.getNetworkId();
 exports._getUtxos = maybe => conn => () =>
   conn.getUtxos().then(res => (res === null ? maybe.nothing : maybe.just(res)));
 
+const getCollateral = conn =>
+  conn.getCollateral !== undefined
+    ? conn.getCollateral
+    : conn.experimental.getCollateral;
+
 exports._getCollateral = maybe => conn => () =>
-  conn.experimental
-    .getCollateral()
+  getCollateral(conn)().then(utxos =>
+    utxos !== null && utxos.length ? maybe.just(utxos) : maybe.nothing
+  );
+
+// Yoroi wallet requires an explicit amount as argument to `getCollateral` and
+// needs it in the form of `getCollateral(amount)` as opposed to CIP-30's
+// `getCollateral(params: { amount: Cbor })`.
+exports._getCollateralYoroi = maybe => conn => () =>
+  conn
+    .getCollateral(3000000)
```

Comment:
This is wrong number. It should be 5.000.000 lovelace. If the wallet fails to respect `maxCollateralInputs`, it's their bug that we should report (I asked Amir to report it upstream)

Reply:
`getCollateral` returns collateral which is bigger than amount. So I thing the most general call should set amount to 0, no?

Also I think PR should be rebased onto mine PR on optional params to prevent future merge conflicts.